src/main/java/net/javadiscord/javabot/systems/staff_commands/RedeployCommand.java

```diff
@@ -39,8 +42,23 @@ public void execute(@NotNull SlashCommandInteractionEvent event) {
 			return;
 		}
 		log.warn("Redeploying... Requested by: " + event.getUser().getAsTag());
-		event.reply("**Redeploying...** This may take some time.").queue();
+		event.reply("Redeploying, this may take a few minutes.").queue();
+		try {
+			Process p = new ProcessBuilder("/bin/sh", Bot.getConfig().getSystems().getRedeployScriptLocation()).start();
+			p.waitFor();
+			String result = new String(p.getInputStream().readAllBytes());
+			if (result.contains("COMPILATION FAILED")) {
+				event.getHook().sendMessage("Compilation failed, redeploy canceled.").queue();
+				log.warn("Redeploy canceled due to compilation error.");
+				return;
+			} else {
+				event.getHook().sendMessage("Compilation successful, restarting...").queue();
+			}
+		} catch (InterruptedException | IOException e) {
+			ExceptionLogger.capture(e, getClass().getSimpleName());
+		}
 		Bot.getMessageCache().synchronize();
+		event.getJDA().shutdown();
 		System.exit(0);
```

Comment:
We may want to move this to its own listener so that it is called when shutting down JDA completed.